addons/portal/static/src/js/portal_signature.js

```diff
@@ -1,93 +1,126 @@
-odoo.define('portal.signature_form', function (require){
-    "use strict";
+odoo.define('portal.signature_form', function (require) {
+    'use strict';
 
-    var ajax = require('web.ajax');
-    var base = require('web_editor.base');
     var core = require('web.core');
-    var Widget = require("web.Widget");
-    var rpc = require("web.rpc");
-
+    var dom = require('web.dom');
+    var NameAndSignature = require('portal.name_and_signature').NameAndSignature;
     var qweb = core.qweb;
+    var Widget = require('web.Widget');
+
+    var _t = core._t;
 
+    /**
+     * This widget is a signature request form. It uses
+     * @see NameAndSignature for the input fields, adds a submit
+     * button, and handles the RPC to save the result.
+     */
     var SignatureForm = Widget.extend({
         template: 'portal.portal_signature',
+        xmlDependencies: ['/portal/static/src/xml/portal_signature.xml'],
         events: {
-            'click #o_portal_sign_clear': 'clearSign',
-            'click .o_portal_sign_submit': 'submitSign',
-            'init #o_portal_sign_accept': 'initSign',
+            'click .o_portal_sign_submit': 'async _onClickSignSubmit',
```

Comment:
This will crash as `async` is still a website feature

Reply:
That's unfortunate, I'll revert it then.